# Probability Drills
## Drill Set 1
<b>1. Calculate the probability of flipping a balanced coin four times and getting each pattern: HTTH, HHHH and TTHH.


A balanced coin = fair coin (i.e. 50% chance heads/tails). Each coin flip is independent, and so:

In [3]:
four_flip_prob = 0.5 ** 4
print('P(HTTH) = P(HHHH) = P(TTHH) = {}%'.format(four_flip_prob*100))

P(HTTH) = P(HHHH) = P(TTHH) = 6.25%


<b>2. If a list of people has 24 women and 21 men, then the probability of choosing a man from the list is 21/45. What is the probability of not choosing a man?

Since our population includes only men or women,

$$P(\textrm{not man}) = 1 - P(\textrm{man})$$

Hence,

In [11]:
not_man_prob = 1 - 21 / 45
print('P(not man) = {}%'.format('%.2f' % (not_man_prob*100)))

P(not man) = 53.33%


<b>3. The probability that Bernice will travel by plane sometime in the next year is 10%. The probability of a plane crash at any time is .005%. What is the probability that Bernice will be in a plane crash sometime in the next year

Bernice flying on a plane and the plane crash happening are independent. Hence,

In [52]:
crash_soon_prob = 0.1*0.00005
print('P(Crash | Travel) = P(Crash) * P(Travel) = {}%'.format('%.4f' % (crash_soon_prob*100)))

P(Crash | Travel) = P(Crash) * P(Travel) = 0.0005%


<b>4. A data scientist wants to study the behavior of users on the company website. Each time a user clicks on a link on the website, there is a 5% chance that the user will be asked to complete a short survey about their behavior on the website. The data scientist uses the survey data to conclude that, on average, users spend 15 minutes surfing the company website before moving on to other things. What is wrong with this conclusion?

The problem with this conclusion is the premise that the survey is given for link clicks. The more you click the links the more likely it is that you are a person who has spent a longer time on the website. Of course, there is also the chance that you are a bot spamming links...

## Drill Set 2

<i>A diagnostic test has a 98% probability of giving a positive result when applied to a person suffering from Thripshaw's Disease, and 10% probability of giving a (false) positive when applied to a non-sufferer. It is estimated that 0.5 % of the population are sufferers. Suppose that the test is now administered to a person whose disease status is unknown. Calculate the probability that the test will:

In [75]:
def bayes_prob(prob_B_given_A, prob_A, prob_B):
    result = prob_B_given_A * prob_A / prob_B
    return "P(A | B) = {}%".format('%.3f' % (result*100))

<b>1. Be positive

My intuition thinks:
$$P(\textrm{positive}) = P(\textrm{im sick})\cdot P(\textrm{positive}) + P(\textrm{not sick})\cdot P(\textrm{positive})$$

In [65]:
print('P(positive (guess)) = {}%' \
      .format('%.2f' % (((0.98)*(0.005)+(0.10)*(1-0.005))*100)))

P(positive (guess)) = 10.44%


This turned out to be correct with the solution file.

<b>2. Correctly diagnose a sufferer of Thripshaw's

My intuition would make me think:

$$P(\textrm{sick }|\textrm{ positive}) = P(\textrm{sick})\cdot P(\textrm{positive})$$

Hence, that would be $$(50\%)\cdot(98\%)=49\%$$

However, using Bayes' Rule,

In [78]:
bayes_prob(0.98, 0.005, 10.44)

'P(A | B) = 0.047%'

My intution was quite off!

<b>3. Correctly identify a non-sufferer of Thripshaw's

Similary to above, I would think probably that this would simply be
$$1-49\%=51\%$$

However! Using Bayes' Rule,

In [82]:
bayes_prob(1-0.1, 1-0.005, 1-0.1044)

'P(A | B) = 99.989%'

<b>4. Misclassify the person

Alright, at this point, we can tell my intuition is going to be off. So i'm not even going to bother:

$$P(\textrm{sick }|\textrm{ negative}) + P(\textrm{not sick }|\textrm{ positive})$$

In [83]:
bayes_prob(1-0.98, 0.005, 1-0.1044) # P(sick | negative)

'P(A | B) = 0.011%'

In [84]:
bayes_prob(0.1, 1-0.005, 0.1044) # P(not sick | postiive)

'P(A | B) = 95.307%'

In [85]:
0.011+95.307

95.318

Hence, the total probability is $95.318\%$ to misclassify a person.